lib/sidekiq/throttled/patches/super_fetch.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+require "sidekiq"
+
+module Sidekiq
+  module Throttled
+    module Patches
+      module SuperFetch
+        # Retrieves job from redis.
+        #
+        # @return [Sidekiq::Throttled::UnitOfWork, nil]
+        def retrieve_work
+          work = super
+
+          if work && Throttled.throttled?(work.job)
+            Throttled.cooldown&.notify_throttled(work.queue)
+            requeue_throttled(work)
+            return nil
+          end
+
+          Throttled.cooldown&.notify_admitted(work.queue) if work
+
+          work
+        end
+
+        private
+
+        # Pushes job back to the head of the queue, so that job won't be tried
+        # immediately after it was requeued (in most cases).
+        #
+        # @note This is triggered when job is throttled. So it is same operation
+        #   Sidekiq performs upon `Sidekiq::Worker.perform_async` call.
+        #
+        # @return [void]
+        def requeue_throttled(work)
+          if work.respond_to?(:local_queue)
+            # if a SuperFetch UnitOfWork, SuperFetch will requeue it using lpush
+            work.requeue
+          else
+            # Fallback to BasicFetch behavior
+            redis { |conn| conn.lpush(work.queue, work.job) }
+          end
+        end
+
+        # Returns list of non-paused queues to try to fetch jobs from.
+        #
+        # @note It may return an empty array.
+        # @return [Array<Array(String, String)>]
+        def active_queues
+          throttled_queues = Throttled.cooldown&.queues&.dup || []
+          super.reject do |queue, _private_queue|
+            # Truthy value means queue is throttled, so we should reject it.
+            throttled_queues.delete(queue)
+          end
```

Comment:
`Throttled.cooldown&.queues` is a copy already - it's totally safe to mutate it.
Either way, I think this will be a bit easier to read:

```suggestion
          throttled_queues = Throttled.cooldown&.queues&.to_a&.to_h { [_1, true] }
          return super unless throttled_queues
          
          super.reject { |queue, _private_queue| throttled_queues[queue] }
```

Reply:
And, probably change `queues_cmd` of BasicFetch to follow the pattern:

``` ruby
        def queues_cmd
          throttled_queues = Throttled.cooldown&.queues
          return super unless throttled_queues

          super - throttled_queues
        end
```